### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by nforro**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,40 @@
+[metadata]
+name = rpm
+description = Shim RPM module.
+long_description = file: README.md
+long_description_content_type = text/markdown
+url = https://github.com/packit/rpm-shim
+author = Red Hat
+author_email = user-cont-team@redhat.com
```
Comment (TomasTomecek) [inline setup.cfg]: let's do `hello@packit.dev` even though it's just an alias for the email above

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+[testenv]
+recreate = true
+usedevelop = true
+sitepackages = false
+# pip < 19 requires setup.py
+deps = pip >= 19
+setenv = PYTHONDONTWRITEBYTECODE=1
+commands =
+    python -c "import rpm"
+    python -m pip check
```
Comment (TomasTomecek) [inline tox.ini:16]: should we also install the shim module? and then ideally `cd /` to make sure the installation works properly

**Review by jpopelka — APPROVED** [review]

**Review by TomasTomecek — COMMENTED** [review]
I am confused now: when I install the module to user sitelib, I can no longer import rpm:
```
// /src is a bindmount of this PR inside a container
[root@a59648fe38f0 src]# pip3 install --user .                                                                                                                                                                                                               
Processing /src      
  Installing build dependencies ... done                                                                                                                                                                                                                     
  Getting requirements to build wheel ... done                                                                                                                                                                                                               
  Installing backend dependencies ... done                                                                                                                                                                                                                   
  Preparing metadata (pyproject.toml) ... done                                                                                                                                                                                                               
Building wheels for collected packages: rpm                                                                                                                                                                                                                  
  Building wheel for rpm (pyproject.toml) ... done                                                                                                                                                                                                           
  Created wheel for rpm: filename=rpm-0.0.post7+g403cc33-py3-none-any.whl size=3497 sha256=bef24c750e69aafdc14073985ae4eecff4472bb3b7d8b03aebcb34b4a43ec97d                                                                                                 
  Stored in directory: /tmp/pip-ephem-wheel-cache-gmnf2sfk/wheels/a0/a1/77/062e875f5b546d9255c4403d3277b732c25c5e17cdbf8d14d3                                                                                                                                
Successfully built rpm                                                                                                                                                                                                                                       
Installing collected packages: rpm
Successfully installed rpm-0.0.post7+g403cc33
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

[root@a59648fe38f0 src]# cd /

[root@a59648fe38f0 /]# python3 -c 'import rpm'                                                                                 
Exception: module 'rpm' has no attribute 'expandMacro'
Exception: module 'rpm' has no attribute 'expandMacro'
Exception: Failed to import system RPM module
```

File: rpm/__init__.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    """
+    for path in get_system_sitepackages():
+        logger.debug(f"Trying {path}")
+        try:
+            if try_path(path):
+                logger.debug("Import successfull")
+                return
+        except Exception as e:
+            logger.error(f"Exception: {e}")
+            continue
```
Comment (TomasTomecek) [inline rpm/__init__.py:86]: hmm, the logic looks okay here: check all paths until one works, otherwise raise ImportError

**Comment by nforro** [comment]
> I am confused now: when I install the module to user sitelib, I can no longer import rpm:

Does the container have RPM python bindings installed?
You can also try to enable debug logging: `python -c 'import logging; logging.basicConfig(level=logging.DEBUG); import rpm'`

File: setup.cfg (hunk 1 above)
Comment (nforro) [reply setup.cfg]: Ok, I'll change that.

File: tox.ini (hunk 2 above)
Comment (nforro) [reply tox.ini:16]: tox does install the shim module before running *commands*, in a virtualenv in a temporary directory (you can run `tox -vv` to see all the details).

**Comment by TomasTomecek** [comment]
yes, the proper rpm bindings were installed

I don't know what went wrong exactly, let's keep it as it is right now and do another "usability testing" round before announcing this module publicly

**Comment by nforro** [comment]
Ok, Zuul is not enabled yet, so let me merge this manually.

**Pull request merged**